scripts/us_bea/states_gdp/import_data.py

```diff
@@ -0,0 +1,127 @@
+# limitations under the License.
+"""
+Pulls data from the US Bureau of Economic Analysis (BEA) on quarterly GDP
+per US state. Saves output as a CSV file.
+
+    Typical usage:
+
+    python3 import_data.py
+"""
+from urllib.request import urlopen
+import pandas as pd
+import io
+import zipfile
+
+US_STATES = ['Alabama', 'Alaska', 'Arizona', 'Arkansas',
+             'California', 'Colorado', 'Connecticut', 'Delaware',
+             'District of Columbia', 'Florida', 'Georgia', 'Hawaii', 'Idaho',
+             'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana',
+             'Maine', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota',
+             'Mississippi', 'Missouri', 'Montana', 'Nebraska', 'Nevada',
+             'New Hampshire', 'New Jersey', 'New Mexico', 'New York',
+             'North Carolina', 'North Dakota', 'Ohio', 'Oklahoma', 'Oregon',
+             'Pennsylvania', 'Rhode Island', 'South Carolina', 'South Dakota',
+             'Tennessee', 'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington',
+             'West Virginia', 'Wisconsin', 'Wyoming']
+
+
+class StateGDPDataLoader:
+    """Pulls GDP state data from BEA.
+
+    Attributes:
+        df: DataFrame (DF) with the cleaned data.
+    """
+    ZIP_LINK = "https://apps.bea.gov/regional/zip/SQGDP.zip"
+    FILE = "SQGDP1__ALL_AREAS_2005_2019.csv"
+
+    def __init__(self):
+        """Downloads the data, cleans it and stores it in instance DF."""
+        # Open zip file from link.
+        resp = urlopen(self.ZIP_LINK)
+
+        # Read the file, interpret it as bytes, and create a ZipFile instance
+        # from it for easy handling.
+        zip_file = zipfile.ZipFile(io.BytesIO(resp.read()))
+
+        # Open the specific desired file (CSV) from the folder, and decode it.
+        # This results in a string representation of the file. Then clean it
+        # by removing " characters inserted in the decoding process.
+        data = zip_file.open(self.FILE).read().decode('utf-8').replace('"', '')
+
+        # Turn the string representation of the file into a DF.
+        data = [line.split(",") for line in data.split("\n")]
+        df = pd.DataFrame(data[1:], columns=data[0])
+
+        # Filters out columns that are not US States (e.g. New England).
+        # TODO(fpernice): Add non-state entities.
+        df = df[df['GeoName'].isin(US_STATES)]
+
+        # Gets columns that represent quarters, e.g. 2015:Q2.
+        all_quarters = [q for q in df.columns if q[:2] == "20"]
+
+        # Convert table from wide to long format.
+        df = pd.melt(df, id_vars=['GeoFIPS', 'Unit'],
+                     value_vars=all_quarters,
+                     var_name='Quarter')
+
+        qtr_month_map = {
+            'Q1':'03',
+            'Q2':'06',
+            'Q3':'09',
+            'Q4':'12'
+        }
+
+        def date_to_obs_date(date):
+            """Converts date format e.g. 2005:Q3 to e.g. 2005-09."""
+            date = date.replace("\r", "")
```

Comment:
you replace carriage return several places. can you apply a whitespace trim to all the fields earlier in the dataframe work?